document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoAggregateExpressionParser.java

```diff
@@ -1,25 +1,59 @@
 package org.hypertrace.core.documentstore.mongo.parser;
 
+import static java.util.Collections.unmodifiableMap;
+import static org.hypertrace.core.documentstore.expression.operators.AggregationOperator.AVG;
+import static org.hypertrace.core.documentstore.expression.operators.AggregationOperator.DISTINCT;
+import static org.hypertrace.core.documentstore.expression.operators.AggregationOperator.MAX;
+import static org.hypertrace.core.documentstore.expression.operators.AggregationOperator.MIN;
+import static org.hypertrace.core.documentstore.expression.operators.AggregationOperator.SUM;
+import static org.hypertrace.core.documentstore.mongo.MongoUtils.getUnsupportedOperationException;
+
+import java.util.EnumMap;
 import java.util.Map;
+import lombok.NoArgsConstructor;
 import org.hypertrace.core.documentstore.expression.impl.AggregateExpression;
-import org.hypertrace.core.documentstore.expression.type.SelectingExpression;
-import org.hypertrace.core.documentstore.parser.SelectingExpressionParser;
-import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.documentstore.expression.operators.AggregationOperator;
+import org.hypertrace.core.documentstore.parser.SelectingExpressionVisitor;
+
+@NoArgsConstructor
+final class MongoAggregateExpressionParser extends MongoSelectingExpressionParser {
+  private static final Map<AggregationOperator, String> KEY_MAP =
+      unmodifiableMap(
+          new EnumMap<>(AggregationOperator.class) {
+            {
+              put(AVG, "$avg");
+              put(DISTINCT, "$addToSet");
```

Comment:
`DISTINCT_COUNT` is not required?

Reply:
Mongo doesn't have a support for DISTINCT_COUNT.

It is internally converted to DISTINCT by the query transformers.